handler/OpenIDLoginHandler.php

```diff
@@ -0,0 +1,281 @@
+
+                if (!empty($settings['logoutUrl']) && isset($settings['clientId'])) {
+                    $request->redirectUrl(
+                        $settings['logoutUrl'] .
+                        '?client_id=' . urlencode($settings['clientId']) .
+                        '&post_redirect_uri=' . urlencode($router->url($request, $context->getPath(), "index")) .
+                        '&id_token_hint=' . urlencode($token)
+                    );
+                }
+            }
+        }
+        $request->redirect(Application::get()->getRequest()->getContext()->getPath(), 'index');
+    }
+
+    /**
+     * Sets user friendly error messages, which are thrown during the OpenID auth process.
+     */
+    private function _setSSOErrorMessages(string $ssoError, TemplateManager $templateMgr, Request $request)
+    {
+        $templateMgr->assign('openidError', true);
+        switch ($ssoError) {
+            case 'connect_data':
+                $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.connect.desc.data');
+                break;
+            case 'connect_key':
+                $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.connect.desc.key');
+                break;
+            case 'cert':
+                $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.cert.desc');
+                break;
+            case 'disabled':
+                $reason = $request->getUserVar('sso_error_msg');
+                $templateMgr->assign('accountDisabled', true);
+                if (isset($reason) && !empty($reason)) {
+                    $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.disabled.with');
+                    $templateMgr->assign('reason', $reason);
+                } else {
+                    $templateMgr->assign('errorMsg', 'plugins.generic.openid.error.openid.disabled.without');
+                }
+                break;
+        }
+
+        $context = $request->getContext();
+        $supportEmail = $context != null ? $context->getSetting('supportEmail') : null;
+        if ($supportEmail) {
+            $templateMgr->assign('supportEmail', $supportEmail);
+        }
+    }
+
+    /**
+     * This function is used
+     *  - if the legacy login is activated via plugin settings,
+     *  - or an error occurred during the Auth process to ensure that the Journal Manager can log in.
+     */
+    private function _enableLegacyLogin(TemplateManager $templateMgr, Request $request)
+    {
+        $sessionManager = SessionManager::getManager();
+        $session = $sessionManager->getUserSession();
+        $context = $request->getContext();
+        $loginUrl = $request->url(null, 'login', 'signIn');
+
+        if (Config::getVar('security', 'force_login_ssl')) {
+            $loginUrl = PKPString::regexp_replace('/^http:/', 'https:', $loginUrl);
+        }
+
+        $templateMgr->assign(
```

Comment:
Could you review the plugin for XSS risks? I see a lot of user-provided data here echoed back to the browser via the `openidLogin.tpl` template without escaping; that looks dangerous to me. (This would also need resolving in the 3.3.0 branch of the plugin.)